src/Ocelot/Multiplexer/MultiplexingMiddleware.cs

```diff
@@ -206,16 +240,17 @@ private static void CopyItemsToNewContext(HttpContext target, HttpContext source
     /// <summary>
     /// Creates a new HttpContext based on the source.
     /// </summary>
-    /// <param name="source">The base http context.</param>
+    /// <param name="source">The base http context.</param>
+    /// <param name="bufferedBody">The body buffered.</param>
     /// <returns>The cloned context.</returns>
-    private static HttpContext CreateThreadContext(HttpContext source)
+    private static async Task<HttpContext> CreateThreadContextAsync(HttpContext source, Stream bufferedBody)
     {
         var from = source.Request;
         var target = new DefaultHttpContext
         {
             Request =
             {
-                Body = from.Body, // TODO Consider stream cloning for multiple reads
+                Body = new MemoryStream(), // WIP: by default it's a NullStream, do we really use MemoryStream?
```

Comment:
Why not Body = await CopyBufferToTargetRequestAsync?

Reply:
that was my first implementation too (defined in the bug) but I hoped the DefaultHttpContext had a already declared buffer. I'd rather rollback.